### ISSUE TITLE
Update typescript from 3 to 4

### ISSUE THREAD

**Issue by johann1301s** [jaredpalmer/tsdx#1147]
TSDX uses typescript version 3.

Many new features are available in version 4. Please update to the latest version.

**Comment by tim-phillips** [comment]
TSDX uses typescript v4

**Comment by johann1301s** [comment]
If you use for example use [Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html), for instance;

```
type World = "world";
 
type Greeting = `hello ${World}`;
```

Compilation fails since it is a TS 4.1 feature.

**Comment by OrkhanAlikhanov** [comment]
@tim-phillips That's not correct. At least with yarn. Rollup gets to use ts 3.9. I have a script that removes that typescript from  local node_modules of tsdx. That makes it use ts 4 installed. Something like this:
```
"prebuild": "rm -rf node_modules/tsdx/node_modules/typescript"
```

**Comment by karlhorky** [comment]
There are a number of other issues in this repository about [the super-old TypeScript 3.9.9](https://github.com/jaredpalmer/tsdx/blob/64c50943905c983c98613944d50c20565efe6691/package.json#L99) inside `tsdx@0.14.1` causing problems for people, where this is documented:

- Yarn Resolutions or npm Overrides: https://github.com/jaredpalmer/tsdx/issues/1149#issuecomment-1272043961
- pnpm Overrides: https://github.com/jaredpalmer/tsdx/issues/926#issuecomment-882102507

**Comment by TheMikeyRoss** [comment]
I think this repo is abandoned by @jaredpalmer

**Comment by Vovan-VE** [comment]
Using a fork [weiran-zsd/dts-cli](https://github.com/weiran-zsd/dts-cli).
